packages/core/saved-objects/core-saved-objects-migration-server-internal/src/core/document_migrator.ts

```diff
@@ -321,9 +314,9 @@ function validateMigrationDefinition(
       throw new Error(
         `Invalid convertToMultiNamespaceTypeVersion for type ${type}. Expected value to be a semver, but got '${convertToMultiNamespaceTypeVersion}'.`
       );
-    } else if (Semver.lt(convertToMultiNamespaceTypeVersion, minimumConvertVersion)) {
+    } else if (convertVersion && Semver.neq(convertToMultiNamespaceTypeVersion, convertVersion)) {
```

Comment:
My gut feeling would have been to keep `8.0.0` as the minimum version and introduces a new `maximumConvertVersion` that we would set either to `8.0.0` or the current version like `8.6.0`. But I'm probably overthinking this.